testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpServerTest.java

```diff
@@ -778,133 +774,62 @@ protected SpanDataAssert assertServerSpan(
-          }
-          assertThat(attrs).containsEntry(getAttributeKey(SemanticAttributes.HTTP_METHOD), method);
+          // client.port is opt-in
+          assertThat(attrs).doesNotContainKey(SemanticAttributes.CLIENT_PORT);
 
-          assertThat(attrs)
-              .containsEntry(getAttributeKey(SemanticAttributes.HTTP_STATUS_CODE), statusCode);
-          if (statusCode >= 500 && SemconvStability.emitStableHttpSemconv()) {
+          assertThat(attrs).containsEntry(SemanticAttributes.HTTP_REQUEST_METHOD, method);
+
+          assertThat(attrs).containsEntry(SemanticAttributes.HTTP_RESPONSE_STATUS_CODE, statusCode);
+          if (statusCode >= 500) {
             assertThat(attrs).containsEntry(HttpAttributes.ERROR_TYPE, String.valueOf(statusCode));
           }
 
-          AttributeKey<String> netProtocolKey =
-              getAttributeKey(SemanticAttributes.NET_PROTOCOL_NAME);
-          if (SemconvStability.emitStableHttpSemconv()) {
-            // only protocol names different from "http" are emitted
-            assertThat(attrs).doesNotContainKey(netProtocolKey);
-          } else if (attrs.get(netProtocolKey) != null) {
-            assertThat(attrs).containsEntry(netProtocolKey, "http");
-          }
-          AttributeKey<String> netProtocolVersionKey =
-              getAttributeKey(SemanticAttributes.NET_PROTOCOL_VERSION);
-          if (attrs.get(netProtocolVersionKey) != null) {
-            assertThat(attrs)
-                .hasEntrySatisfying(
-                    netProtocolVersionKey, entry -> assertThat(entry).isIn("1.1", "2.0"));
-          }
           assertThat(attrs).containsEntry(SemanticAttributes.USER_AGENT_ORIGINAL, TEST_USER_AGENT);
 
-          assertThat(attrs).containsEntry(getAttributeKey(SemanticAttributes.HTTP_SCHEME), "http");
+          assertThat(attrs).containsEntry(SemanticAttributes.URL_SCHEME, "http");
           if (endpoint != INDEXED_CHILD) {
-            if (SemconvStability.emitOldHttpSemconv()) {
-              assertThat(attrs)
-                  .containsEntry(
-                      SemanticAttributes.HTTP_TARGET,
-                      endpoint.resolvePath(address).getPath()
-                          + (endpoint.getQuery() != null ? "?" + endpoint.getQuery() : ""));
-            }
-            if (SemconvStability.emitStableHttpSemconv()) {
-              assertThat(attrs)
-                  .containsEntry(
-                      SemanticAttributes.URL_PATH, endpoint.resolvePath(address).getPath());
-              if (endpoint.getQuery() != null) {
-                assertThat(attrs).containsEntry(SemanticAttributes.URL_QUERY, endpoint.getQuery());
-              }
+            assertThat(attrs)
+                .containsEntry(
+                    SemanticAttributes.URL_PATH, endpoint.resolvePath(address).getPath());
+            if (endpoint.getQuery() != null) {
+              assertThat(attrs).containsEntry(SemanticAttributes.URL_QUERY, endpoint.getQuery());
             }
```

Comment:
Should we add asserts for `URL_FULL` as well?

Reply:
Oh, I see it in the abstract client tests. I think it probably doesn't make as much sense on the server side.

Reply:
Yeah, servers don't emit this attribute.